server/aesdsocket.c

```diff
@@ -230,17 +254,36 @@ void receive_send_data(void* thread_param)
 		close_all();
 		exit (EXIT_FAILURE);
 	}
+		
+		
+	}
+	}while(strchr(thread_func_args->buffer, '\n') == NULL);
+	//received_bytes = buf_loc;
+	thread_func_args->buffer[buf_loc] = '\0';
+	total_bytes += buf_loc; //keep track of total bytes
 
-	int size  = lseek(write_fd,0,SEEK_END);
-	thread_func_args->transmit_buffer = (char *)realloc(thread_func_args->transmit_buffer, size*sizeof(char));
+#ifndef USE_AESD_CHAR_DEVICE
+	size  = lseek(write_fd,0,SEEK_END);
```

Comment:
@Sharath-Jonnala I think you should just delete this line, since `size` isn't used anymore